.github/workflows/build.yaml

```diff
@@ -31,6 +32,16 @@ jobs:
         run: |-
           cargo install cargo-make
 
+      - name: Installing protofetch
+        # run: |-
+        #   cargo install protofetch
+        run: |-
+          cargo install --git https://github.com/knox-networks/protofetch --branch nonzero-error-code
+
+      - name: Setup git dependency
+        run: |-
+          git config credential.helper '!f() { echo username=$EXTERNAL_GIT_USER; echo password=$EXTERNAL_GIT_TOKEN; };f'
```

Comment:
@gshuflin -- I forgot that we already have a `developers@knox-networks.com` GitHub account which exists for this kind of thing... I created a PAT for this account that should give repo access to any repo in the `knox-networks` organization as well as public repos. I saved the PAT as an organizational GHA secret: `PROTOFETCH_GITHUB_TOKEN`. You should be able to hit it within a GHA workflow context by using `{{ secrets.PROTOFETCH_GITHUB_TOKEN }}`.

Once you work out the specifics here, I'd like to see the protofetch code moved into and documented as a reusable action across workflows in `knox-networks/github-actions`. That way you won't have to copy/past so much code between projects. Hit me up if you need help getting started with that...